adam-core/src/main/scala/org/bdgenomics/adam/rdd/fragment/FragmentRDD.scala

```diff
@@ -205,11 +205,24 @@ case class FragmentRDD(rdd: RDD[Fragment],
    * the hull of the underlying element.
    *
    * @param elem The Fragment to get the region from.
+   * @param stranded True to report stranded data for each Fragment.
    * @return Returns all regions covered by this fragment.
    */
-  protected def getReferenceRegions(elem: Fragment): Seq[ReferenceRegion] = {
-    elem.getAlignments
-      .flatMap(r => ReferenceRegion.opt(r))
-      .toSeq
+  protected def getReferenceRegions(elem: Fragment,
+                                    stranded: Boolean = false): Seq[ReferenceRegion] = {
+    if (stranded) {
+      elem.getAlignments
+        .map(r => {
+          if (r.getReadMapped && r.getReadNegativeStrand != null) {
```

Comment:
IMO, if you call `stranded` on reads with `readNegativeStrand == null`, you should get the `IllegalArgumentException`, not an `unstranded` region.